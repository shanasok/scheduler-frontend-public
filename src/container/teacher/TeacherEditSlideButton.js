import React from 'react';
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import { setEditSlideTrue } from '../../redux/actions/editSlideAction';
import "../../css/Buttons.css";

class TeacherEditSlideButton extends React.Component {

    render() {
        return <div>
            <button
                className="curvedButton"
                onClick={this.props.actions.setEditSlideTrue}>Edit Slide</button>
        </div>
    }
}

const mapStateToProps = state => ({
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setEditSlideTrue
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(TeacherEditSlideButton);