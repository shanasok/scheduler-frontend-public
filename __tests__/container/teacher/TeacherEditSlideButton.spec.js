import React from 'react';
import TeacherEditSlideButton from '../../../src/container/teacher/TeacherEditSlideButton';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {Provider as ReduxProvider} from "react-redux";

const mockStore = configureMockStore()
const mockState = {
    slideEditMode: false,
}
let store = mockStore(mockState)

it('renders correctly when setEditSlide boolean is false', () => {

    const tree = renderer
        .create( <ReduxProvider store={store}><TeacherEditSlideButton/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children).toContain("Edit Slide");
});
