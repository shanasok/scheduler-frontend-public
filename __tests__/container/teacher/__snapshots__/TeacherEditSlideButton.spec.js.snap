// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when setEditSlide boolean is false 1`] = `
<div>
  <button
    className="curvedButton"
    onClick={[Function]}
  >
    Edit Slide
  </button>
</div>
`;
